src/Repository/CreditRepository.php

```diff
@@ -0,0 +1,73 @@
+<?php
+namespace App\Repository;
+
+use Wikimedia\ToolforgeBundle\Service\ReplicasClient;
+
+class CreditRepository {
+
+	public function getDatabaseName( $domain, $family ) {
+		if ( $family === 'wikisource' ) {
+			if ( in_array( $domain, [ 'old', 'www', 'mul' ] ) ) {
+				$dbname = 'sourceswiki_p';
+			} elseif ( $domain === 'zh-min-nan' ) {
+				$dbname = 'zh_min_nanwikisource_p';
+			} else {
+				$dbname = $domain . $family . '_p';
+			}
+		} else {
+			$dbname = $domain . $family . '_p';
+		}
+		return $dbname;
+	}
+	
+	public function getPageCredits( ReplicasClient $replicasClient, $lang, $namespaces, $pages ) : array {
+		$dbname = $this->getDatabaseName( $lang, 'wikisource' );
+
+		$conn = $replicasClient->getConnection( $dbname );
+		$pageNames = [];
+		$namespaceName = "";
+		$splittedPages = [];
+	
+		foreach ( $pages as $page ) {
+			$splitPage = explode( ":", $page, 2 );
+			if ( isset( $splitPage[1] ) ) {
+				$namespaceName = $splitPage[0];
+				$pageNames[] = $splitPage[1];
+				$namespaceId = $namespaces[$namespaceName];
+				$splittedPages[$namespaceId][] = "'" . $splitPage[1] . "'";
+			} else {
+				$pageNames[] = $splitPage[0];
+				$splittedPages[0][] = "'" . $splitPage[0] . "'";
+			}
+		}
+
+		$whereSql = [];
+
+		foreach ( $splittedPages as $namespaceId => $pages ) {
+			$whereSql[] = "( page_namespace = $namespaceId AND page_title IN (" . implode( ",", $pages ) . ")" . ")";
+		}
+		$whereSqlString = "(" . implode( ' OR ', $whereSql ) . ")";
+
+		$stmt = $conn->prepare( "SELECT
+			actor_name,
+			COUNT(rev_id) AS count,
+			(
+			SELECT 1
+			FROM enwikisource_p.user_groups
+			WHERE ug_user = user_id
+				AND ug_group = 'bot'
+			) AS bot
+			FROM
+			enwikisource_p.revision_userindex
+			JOIN enwikisource_p.page ON page_id = rev_page
+			JOIN enwikisource_p.actor_revision ON actor_id = rev_actor
+			JOIN enwikisource_p.user ON user_id = actor_user
```

Comment:
the `enwikisource_p.` prefix should be removed, since the database is set when you call `$replicasClient->getConnection( $dbname );`

Reply:
Just a note that if you need to query a different database, such as `commonswiki`, you'll need to get a separate Connection instance as with `$replicasClient->getConnection( 'commonswiki' )`

Reply:
Done.